plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts

```diff
@@ -0,0 +1,236 @@
+    config.getOptionalString('scaffolder.defaultCommitMessage') || commitSubject
+  }\n\nChange-Id: I${changeId}`;
+  return msg;
+};
+
+/**
+ * Creates a new action that initializes a git repository using the content of the workspace.
+ * and publishes it to a Gitea instance.
+ * @public
+ */
+export function createPublishGiteaAction(options: {
+  integrations: ScmIntegrationRegistry;
+  config: Config;
+}) {
+  const { integrations, config } = options;
+
+  return createTemplateAction<{
+    repoUrl: string;
+    description: string;
+    defaultBranch?: string;
+    gitCommitMessage?: string;
+    gitAuthorName?: string;
+    gitAuthorEmail?: string;
+    sourcePath?: string;
+  }>({
+    id: 'publish:gitea',
+    description:
+      'Initializes a git repository using the content of the workspace, and publishes it to Gitea.',
+    examples,
+    schema: {
+      input: {
+        type: 'object',
+        required: ['repoUrl'],
+        properties: {
+          repoUrl: {
+            title: 'Repository Location',
+            type: 'string',
+          },
+          description: {
+            title: 'Repository Description',
+            type: 'string',
+          },
+          defaultBranch: {
+            title: 'Default Branch',
+            type: 'string',
+            description: `Sets the default branch on the repository. The default value is 'master'`,
+          },
+          gitCommitMessage: {
+            title: 'Git Commit Message',
+            type: 'string',
+            description: `Sets the commit message on the repository. The default value is 'initial commit'`,
+          },
+          gitAuthorName: {
+            title: 'Default Author Name',
+            type: 'string',
+            description: `Sets the default author name for the commit. The default value is 'Scaffolder'`,
+          },
+          gitAuthorEmail: {
+            title: 'Default Author Email',
+            type: 'string',
+            description: `Sets the default author email for the commit.`,
+          },
+          sourcePath: {
+            title: 'Source Path',
+            type: 'string',
+            description: `Path within the workspace that will be used as the repository root. If omitted, the entire workspace will be published as the repository.`,
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          remoteUrl: {
+            title: 'A URL to the repository with the provider',
+            type: 'string',
+          },
+          repoContentsUrl: {
+            title: 'A URL to the root of the repository',
+            type: 'string',
+          },
+          commitHash: {
+            title: 'The git commit hash of the initial commit',
+            type: 'string',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const {
+        repoUrl,
+        description,
+        defaultBranch = 'main',
+        gitAuthorName,
+        gitAuthorEmail,
```

Comment:
```suggestion
        gitAuthorName = config.getOptionalString('scaffolder.defaultAuthor.name'),
        gitAuthorEmail = config.getOptionalString('scaffolder.defaultAuthor.email'),
```

if you want to handle all defaults up here (but that won't handle the empty string case)